nntrainer/models/neuralnet.h

```diff
@@ -292,6 +292,17 @@ class NeuralNetwork : public ml::train::Model {
   int setDataset(const DatasetModeType &dt,
                  std::shared_ptr<ml::train::Dataset> dataset) override;
 
+  /**
+   * @copydoc void forEachLayer(std::function<void(Layer &,
+   * nntrainer::RunLayerContext &), void *user_data> fn);
+   *
+   */
+  void forEachLayer(
+    std::function<void(ml::train::Layer & /**< layer */,
```

Comment:
This interface will be limited to running the layer with ml::train::layer interface. If you want more functionality, `ml::train::Layer` can be `nntrainer::Layer` (it's not just a typecast anymore). Feel free to do later.